kiali/kiali
# Canary versions autodiscover

### Describe the change

Removed `ExternalServices.Istio.IstioCanaryRevision` configuration usage, now Kiali checks if there are Istiod pods with  `istio.io/rev=canary` and `istio.io/rev=default` labels.

### Steps to test the PR

Install Istio default.
Install Istio 1.22.2.
Install bookinfo.
Install Istio 1.22.3.
In Kiali UI you should be able to canary upgrade downgrade your data plane namespaces.
In Mesh page, the piechart is removed for dataplane.
In Mesh page, the piechart is changed to support multiple revisions.

![Screenshot from 2024-08-08 11-39-12](https://github.com/user-attachments/assets/05e86ac2-75e2-4263-859a-6d07a8fa1a2e)
![Screenshot from 2024-08-08 11-39-25](https://github.com/user-attachments/assets/6d0f8b45-27c5-45da-ae7e-e32e2d6b26b4)
![Screenshot from 2024-08-08 11-41-00](https://github.com/user-attachments/assets/d30ec06e-d370-409e-8482-8c36e8b8aa99)
![Screenshot from 2024-08-13 12-43-36](https://github.com/user-attachments/assets/644eb3ed-edce-4e8f-beb7-a1cf6c637015)
![Screenshot from 2024-08-13 12-43-17](https://github.com/user-attachments/assets/ee27a1fb-e2aa-4880-841f-ea58a156e980)



### Automation testing

Unit tests are modified to support listing istiod pods.

### Issue reference

https://github.com/kiali/kiali/issues/7515
